spec/requests/custom_buttons_spec.rb

```diff
@@ -147,6 +183,44 @@
       expect(cb.reload.visibility[:roles]).to eq(['_ALL_'])
       expect(cb.reload.resource_action.ae_attributes).to include("attribute" => "is present")
     end
+
+    it 'updates the existing resource action' do
+      cb.resource_action = ResourceAction.create!(:dialog_id => 4)
```

Comment:
Should this be using the `:resource_action` factory and its `:with_dialog` trait? I'm not a fan of using hard-coded ids in test since they're unlikely to be in the same region that the test is running in.